cmd/kubeadm/app/apis/kubeadm/v1alpha1/defaults.go

```diff
@@ -68,4 +72,17 @@ func SetDefaults_NodeConfiguration(obj *NodeConfiguration) {
 	if obj.CACertPath == "" {
 		obj.CACertPath = DefaultCACertPath
 	}
+	if len(obj.TLSBootstrapToken) == 0 {
+		obj.TLSBootstrapToken = obj.Token
+	}
+	if len(obj.DiscoveryToken) == 0 {
```

Comment:
Just realized that if you do this:
```
kubeadm join --discovery-file foo --token bar
```
this will error out that two methods can't be used (file and token)

So let's make this if check `if len(obj.DiscoveryToken) == 0 && len(obj.DiscoveryFile) == 0 {`, right?
cc @jbeda is my logic right here?